From 0b6b01257ef40ab5691956f6a5d4a4c528007c78 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 03 Jul 2019 10:21:01 +0300
Subject: [PATCH] [SW-0] gaudi: add ltssm h/w bug w/a

During the link negotiation, the h/w may change the LTSSM and will not
preserve the initial mask.
To w/a the problem, always write the correct mask to the LTSSM_EN
register after the link is up.

Change-Id: I36a5a808592ad31de8a5016f7f88c3002e3d3aec
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 0cef19a..762c79c 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -48,6 +48,9 @@
 			continue;
 		}
 	}
+
+	/* W/A H/W bug that may lead to have a wrong mask (0x1) in LTSSM */
+	regmap_writel(3, mmPCIE_AUX_LTSSM_EN);
 }
 
 void preboot_delay(void) {}
